osquery/utils/aws_util.h

```diff
@@ -201,6 +201,10 @@ Status makeAWSClient(std::shared_ptr<Client>& client,
   } else {
     client_config.region = region;
   }
+
+  // Setup any proxy options on the config
+  setAWSProxy(client_config);
```

Comment:
I prefer this to be gated somehow, i.e. if user doesn't specify any proxying don't call this function. Maybe introduce a new flag `aws_enable_proxy`.